include/libcgi/cgi.h

```diff
@@ -95,48 +671,147 @@ extern formvars *sess_list_start;
+ *
+ * @note We are opening and closing the session file every time this
+ * function is called... ( I/O ^ 1000000 :-/ )
+ *
+ * @param[in] name Variable name
+ * @param[in] value Variable value
  *
- *	@note	This is some kind of workaround, you still have to call
- *			cgi_end(), but maybe not cgi_destroy() … :-/
+ * @see cgi_session_alter_var(), cgi_session_unregister_var()
  *
- *	@see	https://github.com/rafaelsteil/libcgi/issues/25
+ * @return True in case of success, false on error.
  */
-void cgi_session_free( void );
+extern int cgi_session_register_var(const char *name, const char *value);
 
 /**
- *	The version of this library.
+ * Alter session variable value.
  *
- *	@return	Version string.
+ * Change session variable 'name' value to data pointer by 'new_value'.
+ *
+ * @param[in] name Session variable name to change
+ * @param[in] new_value New session variable value
+ *
+ * @see cgi_session_register_var(), cgi_session_unregister_var()
+ *
+ * @return True in case of success, false on error.
  */
-const char *cgi_version( void );
+extern int cgi_session_alter_var(const char *name, const char *new_value);
 
 /**
- *	Send a HTTP header 'Location:' with the uri provided together with a
- *	pseudo header 'Status:' with some HTTP status code. See RFC 3875,
- *	section 6.3 (Response Header Fields) for details on how this is
- *	supposed to work.
+ * Searches for determined session variable.
+ *
+ * @see cgi_session_var()
  *
- *	@see	https://tools.ietf.org/html/rfc3875#section-6.3
+ * @param[in] name Session variable name to search
  *
- *	@param[in]	status_code	HTTP status code number, pick it from our enum type
- *	@param[in]	uri			URI to be sent with the Location: header
+ * @return 1 (true) if variable is registered, 0 (false) if not
  */
-void cgi_redirect_status( enum cgi_http_status_code status_code,
-		const char *uri );
+extern int cgi_session_var_exists(const char *name);
+
+/**
+ * Unregister some session variable.
+ * @param name Session variable name to unregister
+ * @see cgi_session_var_exists(), cgi_session_register_var()
+ */
+extern int cgi_session_unregister_var(char *name);
+
+/**
+ * Starts a new session.
+ *
+ * This function is responsible for starting and creating a new
+ * session. It must be called before any other session function, and
+ * every time before any HTML header has sent.
+ *
+ * @see session_destroy()
+ *
+ * @return True aka 1 in case of success, false aka 0 otherwise.
+ */
+extern int cgi_session_start();
+
+/**
+ * Free all remaining things explicitly or implicitly allocated by a
+ * session.
+ *
+ * @note This is some kind of workaround, you still have to call
+ * cgi_end(), but maybe not cgi_destroy() … :-/
+ *
+ * @see https://github.com/rafaelsteil/libcgi/issues/25
+ */
+void cgi_session_free( void );
+
+/**
+ * ToDo: Check where this is coming from as it has no implementation
+ * @param seconds
+ */
+extern void cgi_session_set_max_idle_time(unsigned long seconds);
```

Comment:
I discovered that this function prototype does not have a matching implementation. Is this still needed?